x/gamm/pool-models/stableswap/amm_test.go

```diff
@@ -42,10 +49,18 @@ func TestCFMMInvariant(t *testing.T) {
 	}
 
 	for _, test := range tests {
+		// two-asset stableswap tests
 		k0 := cfmmConstant(test.xReserve, test.yReserve)
 		xOut := solveCfmm(test.xReserve, test.yReserve, test.yIn)
 		fmt.Println(xOut)
 		k1 := cfmmConstant(test.xReserve.Sub(xOut), test.yReserve.Add(test.yIn))
 		decApproxEq(t, k0, k1, kErrTolerance)
+
+		// multi-asset stableswap tests
+		k2 := cfmmConstantMulti(test.xReserve, test.yReserve, test.uReserve, test.wSumSquares)
+		xOut2 := solveCfmmMulti(test.xReserve, test.yReserve, test.wSumSquares, test.yIn)
+		fmt.Println(xOut2)
+		k3 := cfmmConstantMulti(test.xReserve.Sub(xOut2), test.yReserve.Add(test.yIn), test.uReserve, test.wSumSquares)
+		decApproxEq(t, k2, k3, kErrTolerance)
```

Comment:
@ValarDragon this is the specific test that is still failing (as discussed on slack)